### DISCUSSION TITLE
Does `minio server /mnt/volume1/ /mnt/volume2/` produce storage of sum size volume1+volume2?

### DISCUSSION THREAD

**Discussion by shoce** [minio/minio#17927] [Q&A]
I have two volumes connected to my virtual machine mounted on `/mnt/volume1/` and `/mnt/volume2/`, each volume of 300GB size. Does starting `minio server /mnt/volume1/ /mnt/volume2/` result in minio storage of size approximately 600GB or the volumes will have just the same data mirrored resulting in 300GB storage?

I checked documentation on `minio server` but could not understand this.

**Comment by shoce** [comment]
I had single node with single drive 300GB 85% full, I had two additional 300GB drives so I started a new minio instance with the two empty drives on the same host and copied all data from the previous instance to the new instance with `mc mirror` and I got those both new drives 85% full so it resulted in the same data mirrored on the two new drives. Found out that expanding is possible only with adding more server pools and restarting all the pools at the same time - so no zero-downtime expansion.

**Reply by klauspost** [reply]
Whether you get RAID0 (combined) or RAID1 (mirror) depends on you erasure config. By default minio will use `EC:1`, which will effectively copy the data. Using `EC:0` will be the same as RAID0.

Expansion is not something you should do more often than every year. All of this is in the docs: https://min.io/docs/minio/linux/operations/install-deploy-manage/expand-minio-deployment.html

**Reply by shoce** [reply]
Sounds good, thanks. I tried creating a new `minio server /data-1/ /data-2/` instance with the two volumes and `MINIO_STORAGE_CLASS_STANDARD` env var set to `EC:0` and started copying data into it but again saw the data is mirrored on the volumes. How can I set the minio server to `EC:0` mode?

**Reply by klauspost** [reply]
The syntax is not correct. Use ellipsis notation. Read the docs.

**Reply by shoce** [reply]
I tried setting `MINIO_STORAGE_CLASS_STANDARD` to `0` and got the message from `minio server`:

```
        MINIO_STORAGE_CLASS_STANDARD: Format "EC:<Default_Parity_Standard_Class>" (e.g. "EC:3"). This sets the number of parity disks for MinIO server in Standard mode. Objects are stored in Standard mode, if storage class is not defined in Put request
```

as it said the format is "EC:<Default_Parity_Standard_Class>" (e.g. "EC:3"). Could not find anything more specific about ellipsis notation at https://min.io/docs/minio/linux/reference/minio-server/minio-server.html 

Found only this at https://min.io/docs/minio/linux/reference/minio-server/minio-server.html#envvar.MINIO_STORAGE_CLASS_STANDARD :
```
The minimum supported value is 0, which indicates no erasure coding protections. These deployments rely entirely on the storage controller or resource for availability / resiliency.
```

**Reply by shoce** [reply]
So my guess that by "ellepsis notation" is meant `minio server /mnt/volume{1...2}/` instead of `minio server /mnt/volume1/ /mnt/volume2/`. I tried it together with `MINIO_STORAGE_CLASS_STANDARD: "EC:0"` and again I see the data mirrored on the volumes, i.e. RAID1 mirror mode, not RAID0 combined mode. 

Regarding expansion recommendation, I totally agree, I am going to have a new volume that should last me a year minimum, but I am trying to make the procedure of expansion ready documented and tested now.

**Reply by cocoshe** [reply]
> So my guess that by "ellepsis notation" is meant `minio server /mnt/volume{1...2}/` instead of `minio server /mnt/volume1/ /mnt/volume2/`. I tried it together with `MINIO_STORAGE_CLASS_STANDARD: "EC:0"` and again I see the data mirrored on the volumes, i.e. RAID1 mirror mode, not RAID0 combined mode.
> 
> Regarding expansion recommendation, I totally agree, I am going to have a new volume that should last me a year minimum, but I am trying to make the procedure of expansion ready documented and tested now.

Hey, do you solve the problem? I also tried to set `MINIO_STORAGE_CLASS_STANDARD` as `EC:0`, and the `mc info` show the status:
![pic1](https://github.com/user-attachments/assets/b75681f3-b59e-40d6-ae2a-3ca6aa961534)
But when I check the disk usage with `du -sh /path/to/data` of my two nodes, the size is the same, so it is still mirror storage.

**Reply by shoce** [reply]
hey @cocoshe, your configuration looks different to what i tried to make. i was trying to use two disks on the same server. you are trying to use two servers with one disk on each. 

regarding my setup, i just gave up and used one big disk. not sure how i will expand it in future. my idea was to attach more disks to the same server but seems like it does not work with minio server.

**Reply by cocoshe** [reply]
> hey @cocoshe, your configuration looks different to what i tried to make. i was trying to use two disks on the same server. you are trying to use two servers with one disk on each.

Yes, it is the difference of our configurations. However, I think it should be the the same goal, to **make use of the space of disks**. After read the doc, I think minio make a trade-off between space(raw disk space) and safety(data loss). The detailed disk usage can be calculated [here](https://min.io/product/erasure-code-calculator), with bigger `EC`, the data is safer, but the total disk space that can be used to save data would be less than the sum of all disk space, since some of the space is used for safety.

According to the calculation, if we build the distributed minio with more nodes(or drives), and with a small `EC`, maybe we can make use of all the disk space, and the small `EC` means less nodes are allowed to be broken when restoring the data.
![img0](https://github.com/user-attachments/assets/608ea51b-2d41-4081-a6a1-717392f35082)
But I only have 2 node for now, so I haven't try it yet.

However, I set `EC:0` successfully (according to the `mc info`), and the `Drives Usage` show the `total: 10TiB` is exactly my (node1+node2) disk. I also try to use the default setting in my case, which is `EC:1`, and the `Drives Usage` is about `total: 6GiB`, which is just one of my node's drive. The difference of total disk from `mc info` indicates I set the `EC:0` successfully now, But I check the disk usage with `du -sh` on the 2 nodes, they are still mirror storage, I am really confused......

**Comment by dijey099** [comment]
After searching for solution to extend my Minio volume without using the mirrored way. I can use `minio server /mnt/vol1 /mnt/vol2`
How ? By analyzing the log of the service, I found that my deployment created a directory .minio.sys in my mount point (/mnt/vol1), where it says that the deployment has 1 disk pool... So I deleted the directory (/mnt/vol1/.minio.sys) and restart the service. It created new .minio.sys in all mount point and in Minio console, I can find out that 2 drives are used
